### PR TITLE
Experimental Separator Component

### PR DESCRIPTION
**Pull request by natalieethell**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ npm run change`

#### Description of changes

This PR adds the experimental separator component. There are two main variations: vertical and horizontal. The default is horizontal, and the vertical variation can be obtained by passing in the `vertical` prop. A developer can also specify how they would like to align the text, using the `alignText` prop. Below I've included screenshots of the demo page in experiments.

image
image

Some thing(s) to note:
* The height of the vertical separator is currently set to 200px. I'm not sure if we'd want some sort of default value here or if we'd like it to inherit the parent's height, so let me know if you have any thoughts on that!
* The background color of the text is set to white, however, I believe that we'd like this to always match the background color of the web app the separator is placed in. Would we want this to be a prop? Or is a white default okay?
* I set the color of the line to `theme.palette.neutralLight`. Let me know if we'd like to use a different color from palette instead!

### CODE REVIEW

File: packages/experiments/src/components/Separator/Separator.types.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+  /**
+   * Accept custom classNames.
+   */
+  className?: string;
+
+  /**
+   * How to align text in the separator.
+   */
+  alignText?: 'start' | 'center' | 'end';
```
Comment (KevinTCoughlin) [inline packages/experiments/src/components/Separator/Separator.types.ts]: 👍

File: packages/experiments/src/components/Separator/Separator.types.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
+import * as React from 'react';
+import { SeparatorBase } from './Separator.base';
+import { IStyleFunctionOrObject, IRefObject } from '../../Utilities';
+import { IStyle, ITheme } from '../../Styling';
+
+export interface ISeparator {}
+
+export interface ISeparatorProps extends React.Props<SeparatorBase> {
+  /**
+   * Optional callback to access the IChiclet interface. Use this instead of ref for accessing
```
Comment (KevinTCoughlin) [inline packages/experiments/src/components/Separator/Separator.types.ts]: nit: `IChiclet` => `ISeparator`

File: packages/experiments/src/components/Separator/Separator.styles.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+      },
+      className
+    ],
+    text: [
+      {
+        position: 'relative',
+        display: 'inline-block',
+        fontSize: '16px',
+        padding: '0 20px',
+        background: 'white'
```
Comment (KevinTCoughlin) [inline packages/experiments/src/components/Separator/Separator.styles.ts]: Would `white` look strange if this were used on say a "dark theme"?

File: packages/experiments/src/components/Separator/Separator.types.ts [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+   */
+  alignText?: 'start' | 'center' | 'end';
+
+  /**
+   * Optional class for separator.
+   */
+  className?: string;
+}
+
+export interface ISeparatorStyleProps {
```
Comment (KevinTCoughlin) [inline packages/experiments/src/components/Separator/Separator.types.ts]: You may consider using `Pick<T, K extends keyof T>` here where `T` is `ISeparatorProps` and the keys are common fields if it helps you.

I just remember it from our Styling Wiki: https://github.com/OfficeDev/office-ui-fabric-react/wiki/Component-Styling#icomponentstyleprops

File: packages/experiments/src/components/Separator/Separator.types.ts [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+   */
+  alignText?: 'start' | 'center' | 'end';
+
+  /**
+   * Whether the separator is vertical.
+   */
+  vertical?: boolean;
+}
+
+export interface ISeparatorStyles {
```
Comment (KevinTCoughlin) [inline packages/experiments/src/components/Separator/Separator.types.ts]: IIRC we recommend all `IComponentStyles` be `required` properties.

https://github.com/OfficeDev/office-ui-fabric-react/wiki/Component-Styling#icomponentstyles

**Comment by natalieethell** [comment]
Changed the vertical separator to fill its container's height by default, fixed the background color of text to work with themes by changing it to `theme.semanticColors.bodyBackground`, changed ISeparatorStyleProps to use Pick, and made all ISeparatorStyles required.

**Pull request merged**